src/modules/communication/internals/sendgrid-service.ts

```diff
@@ -0,0 +1,34 @@
+import mail, { MailDataRequired } from '@sendgrid/mail';
+import { SendEmailParams, ThirdPartyServiceError } from '../types';
+import ConfigService from '../../config/config-service';
+import EmailParams from './sendgrid-email-params';
+
+export default class SendGridService {
+  public static initializeService(): void {
+    mail.setApiKey(
+      ConfigService.getStringValue('sendgridApiKey'),
+    );
+  }
+
+  public static async sendEmail(params: SendEmailParams): Promise<void> {
+    EmailParams.validate(params);
+    const msg: MailDataRequired = {
+      to: params.recipient.email,
+      from: {
+        email: params.sender.email,
+        name: params.sender.name,
+      },
+      templateId: params.templateId,
+      dynamicTemplateData: {
+        ...params.templateData,
+        domain: ConfigService.getStringValue('webAppHost'),
+      },
+    };
+
+    try {
+      await mail.send(msg);
+    } catch (e) {
+      throw new ThirdPartyServiceError('Email service unavailable.');
```

Comment:
Please log the third party error message so we can get notified and proactively investigate about it.